### DISCUSSION TITLE
Locking register value

### DISCUSSION THREAD

**Discussion by shuffle2** [NationalSecurityAgency/ghidra#6629] [Q&A]
It's quite common for programs to setup some register values during initialization which are then considered constant for the life of the program. Or, they use constant register values setup by OS or other external environment.

When using the "Set register values..." UI, a user can set register values for a given address range. However, later interactions with ghidra may have side effects which improperly change the user-defined register values. Is there a way to prevent ghidra changing user-defined register values?

**Comment by ghidra1** [comment]
Register values which you feel are constant across the entire program and have been explicitly set by the user should probably not be changed by an analyzer.  If you have specific cases where an analyzer changes a register value which has previously been set , please open an Issue citing specific details (i.e., sample binary, processor and registers affected).

**Reply by shuffle2** [reply]
Thanks. I think the root cause was a bug in my sleigh code which was causing some instruction sequences to be converted to CALL_RETURN, and the successive address would have the registers reset to UNKNOWN state.

I also realized that "default" register values can be setup in cspec with `tracked_set`.
